porch/controllers/rootsyncrollout/api/v1alpha1/rootsyncrollout_types.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RootSyncRolloutSpec
+type RootSyncRolloutSpec struct {
+	Targets []ClusterTargetSelector `json:"targets,omitempty"`
+
+	Packages PackageSelector `json:"packages,omitempty"`
+}
+
+type ClusterTargetSelector struct {
+	ClusterType ClusterType `json:"clusterType,omitempty"`
+
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+}
+
+type ClusterType string
+
+const (
+	ConfigConnectorCluster ClusterType = "configConnector"
+	LocalCluster           ClusterType = "local"
+)
+
+type PackageSelector struct {
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
I guess we need to go cross-namespace on either packages or targets, and I agree that packages is less dangerous!

Reply:
I think we need to allow for cross-namespace. I added this partly as a way to allow for restricting the controller to target a subset of packages, given that package revisions do not yet support labels.